### PR TITLE
#3260 - CAS Integration 3A - Create new Supplier and Site - Name Fix And Error Log Improvement

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Fixed the order that first and last names were provided to the formatter.
- Added the HTTP bad request logs for better troubleshooting while the final error handling ticket is pending.

### Sample logged error (Invalid SIN).
image

### CODE REVIEW

File: sources/packages/backend/libs/integrations/src/cas/cas.service.ts [hunk 1]
```diff
@@ -128,8 +133,8 @@ export class CASService {
       const config = await this.getAuthConfig();
       const submittedData: CreateSupplierAndSiteSubmittedData = {
         SupplierName: formatUserName(
-          supplierData.lastName,
           supplierData.firstName,
+          supplierData.lastName,
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/cas/cas.service.ts:137]: 👍

File: sources/packages/backend/libs/integrations/src/cas/cas-formatters.ts [hunk 2]
```diff
@@ -13,10 +13,9 @@ const CAS_CITY_MAX_LENGTH = 25;
  * @returns formatted full name.
  */
 export function formatUserName(firstName: string, lastName: string): string {
-  const formattedName = `${lastName}, ${firstName}`.substring(
-    0,
-    CAS_SUPPLIER_NAME_MAX_LENGTH,
-  );
+  const formattedName = `${lastName}, ${firstName}`
+    .substring(0, CAS_SUPPLIER_NAME_MAX_LENGTH)
+    .trim();
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/cas/cas-formatters.ts:18]: 👍

**Review by dheepak-aot — APPROVED** [review]
Looks good 👍

File: sources/packages/backend/libs/integrations/src/cas/_tests_/unit/cas-test.utils.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      expires_in: 3600,
+    },
+  });
+}
+
+/**
+ * Initializes the service under test.
+ * @returns array of mocked services.
+ */
+export async function initializeService(): Promise<
```
Comment (guru-aot) [inline sources/packages/backend/libs/integrations/src/cas/_tests_/unit/cas-test.utils.ts:41]: 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot

**Pull request merged**